modules/nf-scil/registration/synthregistration/main.nf

```diff
@@ -0,0 +1,64 @@
+process REGISTRATION_SYNTHREGISTRATION {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "freesurfer/synthmorph:latest"
+    containerOptions "--entrypoint ''"
+
+    input:
+    tuple val(meta), path(moving), path(fixed)
+
+    output:
+    tuple val(meta), path("*__output_warped.nii.gz"), emit: warped_image
+    tuple val(meta), path("*__deform_warp.nii.gz"), emit: deform_transform
+    tuple val (meta), path("*__init_warp.lta"), emit: init_transform
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def init = task.ext.init ? "-m " + task.ext.init : "-m affine"
+    def warp = task.ext.warp ? "-m " + task.ext.warp : "-m deform"
+    def header = task.ext.header ? "-H" : ""
+    def threads = task.ext.threads ? "-j " + task.ext.threads : ""
+    def gpu = task.ext.gpu ? "-g" : ""
+    def lambda = task.ext.lambda ? "-r " + task.ext.lambda : ""
+    def steps = task.ext.steps ? "-n " + task.ext.steps : ""
+    def extent = task.ext.extent ? "-e " + task.ext.extent : ""
+    def weight = task.ext.weight ? "-w " + task.ext.weight : ""
+
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+
+    mri_synthmorph ${init} -t ${prefix}__init_warp.lta $moving $fixed
+    mri_synthmorph ${warp} ${gpu} ${lambda} ${steps} ${extent} ${weight} -i ${prefix}__init_warp.lta  -t ${prefix}__deform_warp.nii.gz -o ${prefix}__output_warped.nii.gz $moving $fixed
```

Comment:
I was wondering, is the command able to support multiple transforms ? Just thinking that the module should be able to take a sequence of transforms (not just affine+warp),. Not worth it if the whole script needs a redo though.

Reply:
Not sure I fully understand. There is a "rigid" model also, but initialization with it seems to break the registration, which is consstent with the doc: "Deformable assumes prior affine alignment or initialization with -i". The flag "-m joint" would work so it doesn't require successive affine + deform run: "Joint includes affine and deformable but differs from running both in sequence in that it applies the deformable step in an affine mid-space to guarantee symmetric joint transforms." But building it this way allows for producing a affine only transform also. The warp output of the second synthmorph run generates a combined warp field (of affine + deform).

But if there were let's say a second initialzation (rigid + affine + deform), synthmorph always combine successive registration with any initialization.

Reply:
We'll go with what you have now, a single sequence of transforms. The freesurfer script doesn't seem to be made for this and adapting to it would take too much work. I'm happy with what you have right now anyway